modules/packages/LinearAlgebra.chpl

```diff
@@ -221,6 +222,267 @@ class LinearAlgebraError : Error {
     }
 }
 
+/*
+This class promotes lazy computation of required
+Helper matrices to evaluate matrix exponential.
+
+High precision of expm requires smaller computations,
+hence we do a lazy computation depending on the norm value
+of the matrix.
+
+*/
+class ExpmPadeHelper {
+  type t;
+  var A : t;
+  var A2 : t;
+  var A4 : t;
+  var A6 : t;
+  var A8 : t;
+  var A10 : t;
+  var Exactd4 : real;
+  var Exactd6 : real;
+  var Exactd8 : real;
+  var Exactd10 : real;
+  var Approxd4 : real;
+  var Approxd6 : real;
+  var Approxd8 : real;
+  var Approxd10 : real;
+  var ident : t;
+  var use_exact_onenorm : bool;
```

Comment:
Stylistic nit: you use varied naming strategies.  It'd be good to unify to camelCase for all field and method names (we're in the process of writing up a [style guide](https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/StandardModuleStyle.rst) for our modules, which I've linked)